modules/core/src/v2/coins/eos.ts

```diff
@@ -0,0 +1,219 @@
+  }
+
+  getFamily(): string {
+    return 'eos';
+  }
+
+  getFullName(): string {
+    return 'EOS';
+  }
+
+  getBaseFactor(): number {
+    return 1e4;
+  }
+
+  /**
+   * Flag for sending value of 0
+   * @returns {boolean} True if okay to send 0 value, false otherwise
+   */
+  valuelessTransferAllowed(): boolean {
+    return true;
+  }
+
+  /**
+   * Generate secp256k1 key pair
+   *
+   * @param seed - Seed from which the new keypair should be generated, otherwise a random seed is used
+   */
+  generateKeyPair(seed?: Buffer): KeyPair {
+    if (!seed) {
+      // An extended private key has both a normal 256 bit private key and a 256
+      // bit chain code, both of which must be random. 512 bits is therefore the
+      // maximum entropy and gives us maximum security against cracking.
+      seed = crypto.randomBytes(512 / 8);
+    }
+    const extendedKey = utxoLib.HDNode.fromSeedBuffer(seed);
+    const xpub = extendedKey.neutered().toBase58();
+    return {
+      pub: xpub,
+      prv: extendedKey.toBase58(),
+    };
+  };
+
+  /**
+   * Return boolean indicating whether input is valid public key for the coin.
+   *
+   * @param pub - the pub to be checked
+   */
+  isValidPub(pub: string): boolean {
+    try {
+      utxoLib.HDNode.fromBase58(pub);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  /**
+   * Return boolean indicating whether input is valid seed for the coin
+   *
+   * @param prv - the prv to be checked
+   */
+  isValidPrv(prv: string): boolean {
+    try {
+      utxoLib.HDNode.fromBase58(prv);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  /**
+   * Evaluates whether a memo is valid
+   *
+   * @param memo - the memo to be checked
+   */
+  isValidMemo = function(memo: string): boolean {
+    return memo && memo.length <= 256;
+  };
+
+  /**
+   * Return boolean indicating whether a memo id is valid
+   *
+   * @param memoId - the memo id to be checked
+   */
+  isValidMemoId = function(memoId: string): boolean {
+    if (!this.isValidMemo(memoId)) {
+      return false;
+    }
+
+    let memoIdNumber;
+    try {
+      memoIdNumber = new BigNumber(memoId);
+    } catch (e) {
+      return false;
+    }
+
+    return memoIdNumber.gte(0);
+  };
+
+  /**
+   * Process address into address and memo id
+   * @param address - the address
+   */
+  getAddressDetails(address: string): AddressDetails {
+    const destinationDetails = url.parse(address);
+    const queryDetails = querystring.parse(destinationDetails.query);
+    const destinationAddress = destinationDetails.pathname;
+
+    // EOS addresses have to be "human readable", which means start with a letter, less than 12 characters and only a-z1-5., i.e.mtoda1.bitgo
+    // source: https://developers.eos.io/eosio-cpp/docs/naming-conventions
+    if (!/^[a-z][a-z1-5.]*$/.test(destinationAddress) || destinationAddress.length > EOS_ADDRESS_LENGTH) {
+      throw new Error(`invalid address: ${address}`);
```

Comment:
Should throw an `InvalidAddressError` since we already have that error type defined